axum/CHANGELOG.md

```diff
@@ -12,6 +12,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   Use `axum::middleware::from_extractor` instead ([#1077])
 - **breaking:** Allow `Error: Into<Infallible>` for `Route::{layer, route_layer}` ([#924])
 - **breaking:** `MethodRouter` now panics on overlapping routes ([#1102])
+- **breaking:** Nested `Router`s will no longer delegate to the outer `Router`'s
+  fallback. Instead you must explicitly set a fallback on the inner `Router` ([#1086])
+- **breaking:** The request `/foo/` no longer matches `/foo/*rest`. If you want
+  to match `/foo/` you have to add a route specifically for that ([#1086])
+- **breaking:** Path params for wildcard routes no longer include the prefix
+  `/`. e.g. `/foo.js` will match `/*filepath` with a value of `foo.js`, _not_
+  `/foo.js` ([#1086])
```

Comment:
I think we should separate out these runtime-breaking changes from the rest of the changes, move them to the top and put a big warning in front that these *must* be read when upgrading.

I've been burned by silent breaking changes being burried in the middle of the changelog in the context of actix-web, and I would really like axum to do better.

Reply:
I agree. I wanna re-organize the changelog for 0.6 into the different areas that changed and give examples of how to migrate code using 0.5. That will also include calling out subtle changes like this. I think I'll leave it as is and do the re-org in a separate PR when we're closer to 0.6.

Reply:
I made an issue about this https://github.com/tokio-rs/axum/issues/1247